### DISCUSSION TITLE
Constraint based on EntityId?

### DISCUSSION THREAD

**Discussion by ian-a-anderson** [flipt-io/flipt#2774] [Q&A] [answered]
I'd like to segment our users based on specific known User IDs that we are passing via the EntityID to Flipt. I thought this would be as simple as filtering by the ID "property" but it doesn't appear that the EntityID is exposed in this way:
<img width="841" alt="image" src="https://github.com/flipt-io/flipt/assets/2657265/abc509a1-c350-454e-82ea-254f5e244020">

I guess we could additionally pass the ID alongside with the evaluation request as an additional bit of context, but this seems redundant since it is already passed as the EntityID. Is there any other way to get at this type of filtering without passing another copy?

**Comment by markphelps** [comment, marked as answer]
@ian-a-anderson Thanks for the post! This sounds very similar to https://github.com/flipt-io/flipt/issues/2354

I think we could add a constraint type of `entityID` maybe that is always a string and looks at the entityID of the request instead of the context?

**Reply by ian-a-anderson** [reply]
That sounds perfect! Yeah I think that this might solve for both use cases. Adding it as a specific constraint type makes sense as this isn't necessarily a "Property", and that might also help resolve some minor confusion (I kind of expected it to work this way OOTB although I can understand why this is not the case).

**Comment by yquansah** [comment]
@ian-a-anderson Hey 👋 

What type of experience were you looking for here UX wise? There are two ways we are thinking about it:

1. Make the `entityID` another constraint type: This means that when you create a segment and subsequently create constraints for the segment you would be able to see an "entityID" type which you can choose and populate values for it
2. Make the `entityID` a part of the rule itself which is what https://github.com/flipt-io/flipt/issues/2354 is raising. Essentially, allowing you to either have a rule target an `entityID` or segments

**Reply by ian-a-anderson** [reply]
1. for sure, as you could target a single value, a list of values (multiple constraints), or logic across a single value (like in my screenshot above, where we want to allow everyone BUT a particular ID).

**Reply by yquansah** [reply]
Nice!! Thank you 🙏
